### ISSUE TITLE
excessive logging from control plane & log levels/verbosity not respected

### ISSUE THREAD

**Issue by jakekeeys** [democratic-csi/democratic-csi#334]
a cursory glance at the code suggests some thought was given to log levels and verbosity but it doesn't seem to have been wired up here or in the `csi-grpc-proxy`
most of the noise appears to come from the `csi-driver` and `csi-proxy`
![2023-10-24-104452_1769x910_scrot](https://github.com/democratic-csi/democratic-csi/assets/22008619/ff4d4114-1eee-4722-aebf-4f4d750a4d78)
and the contents of which appears to be result of idle metric collection for volumes
```

**Comment by travisghansen** [comment]
Yeah, those need cleaned up! I need to pass in the logger to the util classes and then replace console.log. For the proxy maybe just wrap it in a debug env var or something.

**Comment by tobz** [comment]
+1 on this.

The current logging situation makes debugging any problems an unreasonably difficult process.

**Comment by travisghansen** [comment]
That depends on who is debugging ;) usually I am asking for logs and the current data is extremely helpful. The project however is certainly mature enough however at this point that I will indeed implement this differently.

**Comment by tobz** [comment]
Having the current _set_ of logs be emitted by default would be fine... if everything was actually in a consistent log format, with commensurate log levels applied. I'm fine with filtering out unnecessary logs before they're shipped out of the cluster (even if it's slightly suboptimal) but currently, there's no reasonable way (short of hand-crafting patterns to match all of the various logging categories) to actually process them consistently.

Happy to try and submit a PR for some of this (bring solutions, not problems, etc etc) unless you already have something in the works.

**Comment by travisghansen** [comment]
Always happy for contributions but it really shouldn't be overly difficult for me to add this. All of the points are well received and agreed with by my part!
